packages/peregrine/lib/talons/Header/useHeader.js

```diff
@@ -1,21 +1,39 @@
-import { useCallback } from 'react';
+import { useCallback, useEffect } from 'react';
+import { useLocation } from 'react-router-dom';
+
 import { useAppContext } from '@magento/peregrine/lib/context/app';
+import { useDropdown } from '@magento/peregrine/lib/hooks/useDropdown';
 
 export const useHeader = () => {
-    const [
-        { hasBeenOffline, isOnline, isPageLoading, searchOpen },
-        { toggleSearch }
-    ] = useAppContext();
+    const [{ hasBeenOffline, isOnline, isPageLoading }] = useAppContext();
+    const {
+        elementRef: searchRef,
+        expanded: isSearchOpen,
+        setExpanded: setIsSearchOpen,
+        triggerRef: searchTriggerRef
+    } = useDropdown();
+
+    const location = useLocation();
 
     const handleSearchTriggerClick = useCallback(() => {
-        toggleSearch();
-    }, [toggleSearch]);
+        // Toggle the Search input form.
+        setIsSearchOpen(isOpen => !isOpen);
+    }, [setIsSearchOpen]);
+
+    // If we're on the search page, ensure the search textbox is open to begin with.
+    useEffect(() => {
+        if (location.pathname === '/search.html') {
```

Comment:
I'm not a fan of `Header` caring about what page it's on but this is what enables us to remove `searchOpen` and `toggleSearch` from Redux.

Previously, this effect was running on the search page. Because it needs to control whether search is open or not, we had to hoist that state into global app state. With this effect being here instead, we no longer have to share that state.

Reply:
I think we wanted to spike on this before committing, but this could be an Apollo @client variable. You could additionally query for this in AppContext and distribute that way, while also providing an action for toggling; we could safely delete the Redux state/action this way, since we've provided a backwards compatible replacement.

Reply:
I'd be fine with removing this, it's a pretty arcane requirement in the first place. We can probably negotiate it with UX.

If it has to stay, the location check is fine. Eventually we'll want a utility function to abstract the whole "what page are we on?" inference using location and some other data/heuristics.

Reply:
Ended up removing this entirely as Jimmy suggested.